server/controllers/report.js

```diff
@@ -0,0 +1,142 @@
+const db = require('../models');
+const { keyBy } = require('lodash');
+
+function findBySubject({ query: { id } }, res) {
+  return db.report.findAll({
+    where: { subjectId: id },
+    include: [
+      {
+        as: 'reporter',
+        model: db.user
+      }
+    ]
+  })
+    .then(reports => res.send(reports));
+}
+
+function getLastUsersReports({ user: { role, team } }, res) {
+  const query = {
+    attributes: [
+      db.sequelize.literal('DISTINCT ON("report"."subject_id") 1'),
+      'id', 'createdAt'
+    ],
+    order: [['subject_id', 'asc'], ['createdAt', 'desc']],
+    include: [{
+      as: 'subject',
+      model: db.user,
+      attributes: ['id', 'firstName', 'lastName', 'office'],
+      required: true
+    }]
+  };
+
+  if (role !== 'ADMIN') {
+    const [userJoin] = query.include;
+    userJoin.where = { team };
+  }
+
+  return db.report.findAll(query)
+    .then(reports => {
+      const users = [];
+      reports.forEach(({ id, createdAt, subject }) => {
+        const report = {
+          id,
+          createdAt
+        };
+        const user = {
+          id: subject.id,
```

Comment:
Can you find a way to not assign each prop individually but do it in bulk or some other, shorter way?

Reply:
I reworked whole process to 4 lines 👍